contractcourt/channel_arbitrator.go

```diff
@@ -1794,7 +1986,7 @@ func (c *ChannelArbitrator) channelAttendant(bestHeight int32) {
 			// a terminal state.
 			_, _, err = c.advanceState(
 				uint32(closeInfo.SpendingHeight),
-				localCloseTrigger,
+				localCloseTrigger, &closeInfo.CommitSet,
```

Comment:
The commit set is only obtained when the chain watcher sends us message that something has confirmed on chain. We only use the commit set when prepping contract resolutions (and check for nil-ness in other areas). We only prep resolutions once a commitment has confirmed. Therefore we'll only access this when it's non-nil and a commitment has confirmed.

Reply:
But if we crash at this point, will we get the commitment confirmed signal again after restart from chain watcher?

Reply:
Hmm yeah at an initial glance, I thought it would continue to send this information until we mark the channel as closed, which is the case, but we do so before we start to advance our state. Atm, we do so only after we finish writing the contract resolutions here: https://github.com/lightningnetwork/lnd/blob/master/contractcourt/channel_arbitrator.go#L1837

It appears that we may also need to start to write this commitment transaction information, as this state isn't stored in the `ChannelCloseSummary` which is what's available to us once we mark a channel as closed.

Reply:
To make this backwards compatible for old nodes (that are for the case of this scenario, "frozen" in this limbo state), then if this new piece of information isn't present, we just only play the commitment at hand.

Reply:
Can this then combined with that other comment about getting the final commit sets during force close?

Reply:
So not passing them in from chain watcher at all anymore

Reply:
I implemented what I described above in this latest diff. I still prefer passing the information from the chain watcher as IMO it's easier to reason about the consistency of the commitment states at that point since the channel has been closed.

Reply:
Ok, I thought maybe this last change changed your view on this.